src/test/end-to-end/helpers/PuppeteerHelper.js

```diff
@@ -13,18 +13,29 @@ class PuppeteerHelper extends Helper {
 
     async waitForNavigationToComplete(locator) {
         const page = this.helpers[helperName].page;
+        const promises = [];
 
-        const promises = [
-            page.waitForNavigation({timeout: 60000, waitUntil: ['domcontentloaded', 'networkidle0']}) // The promise resolves after navigation has finished
-        ];
+        if (!testConfig.TestForXUI) {
+            promises.push(page.waitForNavigation({timeout: 60000, waitUntil: ['domcontentloaded', 'networkidle0']})); // The promise resolves after navigation has finished
+        }
 
         if (locator) {
             promises.push(page.click(locator));
         }
-
         await Promise.all(promises);
     }
 
+    async clickTab(tabTitle) {
+        const helper = this.helpers[helperName];
+        if (testConfig.TestForXUI) {
```

Comment:
Angular mat-tab is used in XUI which will render dynamically and not through clickable link